### ISSUE TITLE
Error in environment process: Traceback (most recent call last):

### ISSUE THREAD

**Issue by www2171668** [HorizonRobotics/alf#1005]
I meet a problem about sac algorithm. When I run "python -m alf.bin.train --gin_file=sac_fetchreach.gin --root_dir="sac_fetchreach" , it will show error as "ValueError: Configurable 'SacAlgorithm' doesn't have a parameter named 'actor_network_cls'". 

![image](https://user-images.githubusercontent.com/49861957/133020395-d4bbcc34-1cd7-4cb1-9831-84af46c8a38b.png)

I find a recent submit in Sac_fetchreach.gin, SacAlgorithm.actor_network=@actor/ActorDistributionNetwork() has been changed to SacAlgorithm.actor_network_cls=@actor/ActorDistributionNetwork(). In SacAlgorith, actor_network_cls is still been used in __init__. 
In my mind, I think the formmer is correct, but it shows error. Why SacAlgorithm.actor_network_cls shows error ,but SacAlgorithm.actor_network is OK?

**Comment by hnyu** [comment]
Hi, please check your local version of the code. It seems that you're using a quite old version of SAC but a newer version of the gin file. The current SacAlgorithm takes a constructor instead of a constructed object for the actor network creation. You can find its init signature at:

https://alf.readthedocs.io/en/latest/api/alf.algorithms.html#module-alf.algorithms.sac_algorithm

![image](https://user-images.githubusercontent.com/51248379/133032723-c72831dd-0945-4e46-8217-a87aa3dc1805.png)

**Comment by www2171668** [comment]
Mybe I use a old virtualenv and a chaotic version. When I change the newest version and create a new virtualenv, I meet a new problem never seen. 
![image](https://user-images.githubusercontent.com/49861957/133082270-67c5fd45-f8eb-48bf-8a0b-5d55d4fc05db.png)

![image](https://user-images.githubusercontent.com/49861957/133082421-df81e961-7624-4e80-8917-f13be15d3901.png)

...
![image](https://user-images.githubusercontent.com/49861957/133080862-4778edad-38d6-494b-b945-ffe24d1bd619.png)

**Comment by www2171668** [comment]
Mybe I use a old virtualenv and a chaotic version. When I change the newest version and create a new virtualenv, I meet a new problem never seen.

![image](https://user-images.githubusercontent.com/49861957/133082718-5968e779-f758-45bb-a9b0-5c1fd2e24f0e.png)

![image](https://user-images.githubusercontent.com/49861957/133082741-ef8eb562-128a-444b-b4db-58c87bc6c354.png)
...
![image](https://user-images.githubusercontent.com/49861957/133082792-039b34cd-a360-4424-933e-7bd454dcd314.png)
...
![image](https://user-images.githubusercontent.com/49861957/133082862-6fb12a91-ec6d-436f-a4de-5f31f214e114.png)
...
![image](https://user-images.githubusercontent.com/49861957/133082900-8749128a-52b2-4a21-bf6b-29a029508f31.png)

The completed error in the txt.
[error.txt](https://github.com/HorizonRobotics/alf/files/7154175/error.txt)

**Comment by hnyu** [comment]
This is an error from openai Gym, not from ALF, as can be seen below:

```
File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/gym/core.py", line 258, in reset
    return self.observation(observation)
  File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/gym/wrappers/flatten_observation.py", line 15, in observation
    return spaces.flatten(self.observation_space, observation)
  File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/gym/spaces/utils.py", line 30, in flatten
    return np.asarray(x, dtype=np.float32).flatten()
  File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/numpy/core/_asarray.py", line 83, in asarray
    return array(a, dtype, copy=False, order=order)
TypeError: float() argument must be a string or a number, not 'dict'
```

Fixed this bug by upgrading gym's version in ALF, see https://github.com/HorizonRobotics/alf/pull/1007. Basically we need to use a version >=0.15.4.

**Comment by hnyu** [comment]
> This is an error from openai Gym, not from ALF, as can be seen below:
> 
> ```
> File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/gym/core.py", line 258, in reset
>     return self.observation(observation)
>   File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/gym/wrappers/flatten_observation.py", line 15, in observation
>     return spaces.flatten(self.observation_space, observation)
>   File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/gym/spaces/utils.py", line 30, in flatten
>     return np.asarray(x, dtype=np.float32).flatten()
>   File "/home/www2171668/.conda/envs/alfnew/lib/python3.7/site-packages/numpy/core/_asarray.py", line 83, in asarray
>     return array(a, dtype, copy=False, order=order)
> TypeError: float() argument must be a string or a number, not 'dict'
> ```
> 
> Fixed this bug with our custom wrapper, see #1006

**Comment by www2171668** [comment]
Thank you very much !!! (・O・。)
